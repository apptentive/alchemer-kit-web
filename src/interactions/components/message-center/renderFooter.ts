/**
 * Renders a given Message Center footer.
 *
 * @returns {HTMLElement} - The footer DOM Node.
 */
const renderFooter = () => {
  const footer = document.createElement('footer');
  footer.classList.add('apptentive-footer');

  return footer;
};

export default renderFooter;
