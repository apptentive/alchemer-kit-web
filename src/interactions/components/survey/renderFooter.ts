interface IFooterOptions {
  contactUrl?: string;
  contactUrlText?: string;
  termsAndConditionsUrl?: string;
  termsAndConditionsText?: string;
}

/**
 * Renders a given surveys footer.
 *
 * @param {object} configuration - The configuration object to drive the footer creation
 * @returns {HTMLElement} - The DOM Node
 */
const renderFooter = (configuration: IFooterOptions) => {
  const footer = document.createElement('footer');
  footer.className = 'apptentive-survey-footer';

  if (configuration.contactUrl && configuration.contactUrlText) {
    const contact = document.createElement('a');
    contact.className = 'apptentive-survey-footer__link';
    contact.textContent = configuration.contactUrlText;
    contact.href = configuration.contactUrl;
    contact.target = '_blank';

    footer.append(contact);
  }

  if (configuration.termsAndConditionsUrl && configuration.termsAndConditionsText) {
    const termsAndConditions = document.createElement('a');
    termsAndConditions.className = 'apptentive-survey-footer__link';
    termsAndConditions.textContent = configuration.termsAndConditionsText;
    termsAndConditions.href = configuration.termsAndConditionsUrl;
    termsAndConditions.target = '_blank';

    footer.append(termsAndConditions);
  }

  return footer;
};

export default renderFooter;
