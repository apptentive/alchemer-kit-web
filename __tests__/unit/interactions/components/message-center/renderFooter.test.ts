import renderFooter from '../../../../../src/interactions/components/message-center/renderFooter';

describe('renderFooter', () => {
  test('properly renders footer', () => {
    const output = renderFooter();

    expect(output).toMatchSnapshot();
  });
});
