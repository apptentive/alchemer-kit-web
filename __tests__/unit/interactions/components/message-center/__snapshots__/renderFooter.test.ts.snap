// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderFooter properly renders footer 1`] = `
<footer
  class="apptentive-footer"
/>
`;
